omod/src/main/webapp/patientcolors.jsp

```diff
@@ -1,33 +1,37 @@
 <DOCTYPE html>
+
+
+<%@ taglib uri = "http://java.sun.com/jsp/jstl/core" prefix = "c" %>
+
 <html>
 <head>
  <title>patient colors information</title>
  
 </head>
 
 <body>
-<form action="/openmrs-standalone/module/patientcolors/patientcolors.form" method="post">
+<form method="POST" modelAttribute="patientcolorss" action = "/openmrs-standalone/module/patientcolors/patientcolors.form">
 <fieldset>
 <legend><b>Enter patient details</b></legend>
 Select patientName:
-<select name = "patient_id">
-<c:forEach var="patient" items"${patients}">
+<select name="patientId">
+<c:forEach items="${patients}" var="patient">
```

Comment:
Man, it's not clear how you capture the patient name. . . .
How do you intend that this should work?

Reply:
@ruhnaga i tried to refactor code to capture the patient name but i kinda failed, how is my post-redirect-get pattern that i used in the controller to avoid double form submission?